### ISSUE TITLE
Plugin Updater Check does not detect PUC library usage

### ISSUE THREAD

**Issue by ernilambar** [10up/plugin-check#1167]
The `look_for_plugin_updaters()` function in `Plugin_Updater_Check` does not detect plugins that use the YahnisElsts Plugin Update Checker library.

**Example code that is currently not detected:**

```php
$update_checker = YahnisElsts\PluginUpdateChecker\v5\PucFactory::buildUpdateChecker(
	'https://example.com/plugin-info.json',
	__FILE__,
	'my-plugin-slug'
);
```

Current code:
```php
$look_for_regex = array(
	'#\'plugin-update-checker\'#',
	'#WP_GitHub_Updater#',
	'#WPGitHubUpdater#',
	'#class [A-Z_]+_Plugin_Updater#i',
	'#updater\.\w+\.\w{2,5}#i',
	'#site_transient_update_plugins#',
);
```

This pattern uses the `YahnisElsts\PluginUpdateChecker\v5\PucFactory::buildUpdateChecker()` static method to add custom update functionality, which is not permitted.

**Comment by davidperezgar** [comment]
Do you want to make it?
